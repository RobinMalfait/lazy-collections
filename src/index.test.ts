import * as lazy from '.'

it('should export all the things', () => {
  expect(lazy).toMatchSnapshot()
})

it('should be possible to create a chain of actions and combine them in a nice stream', () => {
  let program = lazy.pipe(
    // Triple
    lazy.map((x: number) => x * 3),

    // Even
    lazy.filter((x: number) => x % 2 === 0),

    // Take only the ones below 100 000
    lazy.takeWhile((x: number) => x < 100_000),

    // Take the first 10
    lazy.take(10),

    // To array
    lazy.toArray
  )

  expect(program(lazy.range(1_000, 1_000_000))).toEqual([
    3000,
    3006,
    3012,
    3018,
    3024,
    3030,
    3036,
    3042,
    3048,
    3054,
  ])
})
