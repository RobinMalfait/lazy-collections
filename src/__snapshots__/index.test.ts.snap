// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should export all the things 1`] = `
{
  "at": [Function],
  "average": [Function],
  "batch": [Function],
  "chunk": [Function],
  "compact": [Function],
  "compose": [Function],
  "concat": [Function],
  "delay": [Function],
  "every": [Function],
  "filter": [Function],
  "find": [Function],
  "findIndex": [Function],
  "first": [Function],
  "flatMap": [Function],
  "flatten": [Function],
  "generate": [Function],
  "groupBy": [Function],
  "head": [Function],
  "includes": [Function],
  "join": [Function],
  "map": [Function],
  "max": [Function],
  "mean": [Function],
  "min": [Function],
  "partition": [Function],
  "pipe": [Function],
  "product": [Function],
  "range": [Function],
  "reduce": [Function],
  "reverse": [Function],
  "skip": [Function],
  "slice": [Function],
  "some": [Function],
  "sort": [Function],
  "sum": [Function],
  "take": [Function],
  "takeWhile": [Function],
  "tap": [Function],
  "toArray": [Function],
  "unique": [Function],
  "wait": [Function],
  "where": [Function],
  "windows": [Function],
  "zip": [Function],
}
`;
